### ISSUE TITLE
trak-ball limits

### ISSUE THREAD

**Issue by dualcam** [stella-emu/stella#225]
Trak-ball movement limitations described in -
http://atariage.com/forums/topic/269337-colony-7-2600-in-stella-with-trackball/page-2
I suspect are the result of going off the cursor position.  Instead would want to use the raw mouse motion I believe is xrel/yrel here -
http://wiki.libsdl.org/SDL_MouseMotionEvent?highlight=%28%5CbCategoryStruct%5Cb%29%7C%28CategoryEvents%29

**Comment by sa666666** [comment]
I am waiting on a donation of Trakball and related controllers from the very person that's mentioned in that thread.  I will be in a better position to evaluate what's going on then.

But at first glance, we already are using xrel and yrel from SDL.  It's working for the mouse and paddles, so the work done there just needs to be ported to the trakball controller code too.  And the reason this hasn't been done is again because I don't have all the real controllers to test (and limited time, as usual).

**Comment by dualcam** [comment]
OK, no problem.  Was just thinking about the description of it hitting a barrier, and if it was using the cursor position, that could be the reaching of the screen or window edge.

FYI, I could pass the raw trak-ball input signals, the joystick U/D/L/R states, like what is done with the driving controller, but I think using the mouse seems like a more open solution.

Tom

--------------------------------------------
On Fri, 9/22/17, Stephen Anthony <notifications@github.com> wrote:

 Subject: Re: [stella-emu/stella] trak-ball limits (#225)
 To: "stella-emu/stella" <stella@noreply.github.com>
 Cc: "dualcam" <te27s5@yahoo.com>, "Author" <author@noreply.github.com>
 Date: Friday, September 22, 2017, 1:19 PM
 
 I am waiting on a donation of
 Trakball and related controllers from the very person
 that's mentioned in that thread.  I will be in a better
 position to evaluate what's going on then.
 But at first glance, we already are using xrel and yrel
 from SDL.  It's working for the mouse and paddles, so
 the work done there just needs to be ported to the trakball
 controller code too.  And the reason this hasn't been
 done is again because I don't have all the real
 controllers to test (and limited time, as usual).
 
 —
 You are receiving this because you authored the thread.
 Reply to this email directly, view
 it on GitHub, or mute
 the thread.

**Comment by thrust26** [comment]
I still don't understand the problem. I just played Colony 7 with a (dead cheap) Amiga mouse using a 2600-daptor II and there are no barriers whatsoever.

So what's the problem here?

**Comment by dualcam** [comment]
I didn't actually try it.  Was a offering a possible cause based on what was being described in the AA thread.

Tom

--------------------------------------------
On Mon, 10/2/17, Thomas Jentzsch <notifications@github.com> wrote:

 Subject: Re: [stella-emu/stella] trak-ball limits (#225)
 To: "stella-emu/stella" <stella@noreply.github.com>
 Cc: "dualcam" <te27s5@yahoo.com>, "Author" <author@noreply.github.com>
 Date: Monday, October 2, 2017, 7:29 AM
 
 I still don't understand the
 problem. I just played Colony 7 with a (dead cheap) Amiga
 mouse using a 2600-daptor II and there are no barriers
 whatsoever.
 So what's the problem here?
 
 —
 You are receiving this because you authored the thread.
 Reply to this email directly, view
 it on GitHub, or mute
 the thread.

**Comment by sa666666** [comment]
I'm waiting on a donation of all controller types.  Package has been sent; I will revisit this once I get the hardware.

**Comment by dualcam** [comment]
OK, thanks.  FYI - I opened a bug report just to try to follow "official" procedures as didn't see any exiting one.  Understand you are busy and will get to it as you have time and further don't owe me status updates, but anything you want to share or have a dialog about is welcome.

Tom

--------------------------------------------
On Mon, 10/9/17, Stephen Anthony <notifications@github.com> wrote:

 Subject: Re: [stella-emu/stella] trak-ball limits (#225)
 To: "stella-emu/stella" <stella@noreply.github.com>
 Cc: "dualcam" <te27s5@yahoo.com>, "Author" <author@noreply.github.com>
 Date: Monday, October 9, 2017, 12:58 PM
 
 I'm waiting on a donation of
 all controller types.  Package has been sent; I will revisit
 this once I get the hardware.
 
 —
 You are receiving this because you authored the thread.
 Reply to this email directly, view
 it on GitHub, or mute
 the thread.

**Comment by thrust26** [comment]
All fine. I want to get this fixed (or understood) too.

**Comment by thrust26** [comment]
@sa666666 How should we proceed with this one?

**Comment by sa666666** [comment]
No idea, since I haven't had time to revisit this since the initial report.  I now have all the controllers (trakball, amiga/atarimouse), but no time so far to test.  But I haven't done any research whatsoever.

I think this is definitely a post-6.1 thing. @dualcam has reported several bugs/issues, and IMO the more serious one is being able to map according to controller type (so a key means one thing in joystick mode, another in keypad mode, etc).  That is what we're currently working on (well you, really).  I think we should get this and the associated `PJoystickHandler` stuff finished, and come back to this issue after 6.1.

**Comment by thrust26** [comment]
Ok, though I cannot promise when I will get to this. Holidays are coming soon.

**Comment by thrust26** [comment]
I think I am (finally) starting to understand the problem:
- controllers are connected via 2600-daptor
- the Amiga/Atari mice seem to have a higher resolution than the real trackballs
- with "mouse grab **on**", the cursor moves too slow when using real trackballs
- with "mouse grab **off**", the cursor moves faster but stops moving (at that invisible barrier) when the mouse exits the screen (though: why is there a mouse involved at all???)

Is that the summary of what Swami is writing? I find it a bit hard to follow is description of the problem.

**Comment by thrust26** [comment]
I just test with my own controllers and I cannot reproduce the problem. But I think there might be factors important which I am not aware about.

**Comment by thrust26** [comment]
@dualcam Does the problem still exist with 6.1?
@sa666666 What shall we do with this one?

**Comment by sa666666** [comment]
I guess we need to wait for a response.  At least for a little while.

**Comment by thrust26** [comment]
No response.
